src/transform.py

```diff
@@ -171,6 +187,114 @@ def is_confirmed(self):
+    ):
+        # Sort sessions based on start time, early first
+        all_sessions_sorted = sorted(
+            all_sessions.values(), key=lambda x: (x.start is None, x.start)
+        )
+
+        # Filter out sessions
+        remaining_sessions = [
+            session
+            for session in all_sessions_sorted
+            if session.start is not None
+            and session.start >= submission.end
+            and session.code not in submission.talks_in_parallel
+            and session.code != submission.code
+            and submission.start.day == session.start.day
+            and not submission.submission_type
+            == session.submission_type
+            == "Announcements"
+        ]
+
+        # Add sessions to the list if they are in different rooms
+        seen_rooms = set()
+        unique_sessions = []
+
+        for session in remaining_sessions:
+            if session.room not in seen_rooms:
+                unique_sessions.append(session)
+                seen_rooms.add(session.room)
+
+        # If there is a keynote next, only show that
+        if any(s.submission_type == "Keynote" for s in unique_sessions):
+            unique_sessions = [
+                s for s in unique_sessions if s.submission_type == "Keynote"
+            ]
+
+        # Set the next talks in all rooms
+        submission.talks_after = [session.code for session in unique_sessions]
+
+        # Set the next talk in the same room
+        for session in unique_sessions:
+            if session.room == submission.room:
+                submission.next_talk = session.code
+                break
+
+    @staticmethod
+    def set_talks_before(
+        submission: PretalxSubmission, all_sessions: dict[str, PretalxSubmission]
+    ):
+        # Sort sessions based on start time, late first
+        all_sessions_sorted = sorted(
+            all_sessions.values(),
+            key=lambda x: (x.start is None, x.start),
+            reverse=True,
+        )
+
+        remaining_sessions = [
+            session
+            for session in all_sessions_sorted
+            if session.start is not None
+            and session.code not in submission.talks_in_parallel
+            and session.start <= submission.start
+            and session.code != submission.code
+            and submission.start.day == session.start.day
+            and session.submission_type != "Announcements"
+        ]
+
+        seen_rooms = set()
+        unique_sessions = []
+
+        for session in remaining_sessions:
+            if session.room not in seen_rooms:
+                unique_sessions.append(session)
+                seen_rooms.add(session.room)
+
+        submission.talks_before = [session.code for session in unique_sessions]
+
+        for session in unique_sessions:
+            if session.room == submission.room:
+                submission.prev_talk = session.code
+                break
```

Comment:
There is a big amount of overlap with ``set_talks_after`` in this method. We probably should reduce the code duplication in the future.